### PR TITLE
[DOC release] Document on index block param for each helper

### LINKED ISSUE #12721: Document `{{#each` index block parameter
It is documented in the [guides](https://guides.emberjs.com/v2.2.0/templates/displaying-a-list-of-items/#toc_accessing-an-item-s-code-index-code) but not the [api](http://emberjs.com/api/classes/Ember.Templates.helpers.html#method_each)

### PR DESCRIPTION
**Pull request by Charizard**
[Fixes #12721]

### CODE REVIEW

File: packages/ember-htmlbars/lib/helpers/each.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import decodeEachKey from 'ember-htmlbars/utils/decode-each-key';
   of the base Handlebars `{{#each}}` helper.
 
   The default behavior of `{{#each}}` is to yield its inner block once for every
-  item in an array.
+  item in an array passing the item as the first param to the block.
```
Comment (trek) [inline packages/ember-htmlbars/lib/helpers/each.js]: We try to refer to `param` as `parameter` in prose for the API docs. "first block parameter" is fine.

File: packages/ember-htmlbars/lib/helpers/each.js [hunk 2]
```diff
@@ -36,6 +36,16 @@ import decodeEachKey from 'ember-htmlbars/utils/decode-each-key';
   {{/each}}
   ```
 
+  During iteration, the index of each item in the array is provided as a second block param.
```
Comment (trek) [inline packages/ember-htmlbars/lib/helpers/each.js]: s/param/parameter.

**Comment by Charizard** [comment]
@trek Hey, I have made appropriate changes. Can you check it?

**Comment by trek** [comment]
Perfect! Thank you!

**Pull request merged**